app/main.py

```diff
@@ -1,44 +1,67 @@
+import asyncio
 import time
+from typing import Awaitable, Any
 
 from iot.devices import HueLightDevice, SmartSpeakerDevice, SmartToiletDevice
 from iot.message import Message, MessageType
 from iot.service import IOTService
 
 
-def main() -> None:
+async def run_sequence(*functions: Awaitable[Any]) -> None:
+    for function in functions:
+        await function
+
+
+async def run_parallel(*functions: Awaitable[Any]) -> None:
+    await asyncio.gather(*functions)
```

Comment:
In the `run_parallel` function, you're using `asyncio.gather`. This is a good way to run several tasks in parallel, but be aware that if one of the tasks raises an exception, `gather` will cancel the other tasks and raise the exception immediately. If you want all tasks to run to completion even if some of them fail, you should use `asyncio.gather(..., return_exceptions=True)`.